motionplan/motionPlanner.go

```diff
@@ -381,3 +383,125 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
```

Comment:
[q] should context be included as an argument so that we can cancel this if needed?
I am imagining the following scenario.
As we perform obstacle detection we also perform position polling.
If position polling informs us that we are off plan, we will cancel all contexts and replan.
In this case, the context of the on-going obstacle detection should be cancelled..